File: httpjava/src/main/java/org/stuartaroth/httpjava/services/id/UniqueIdService.java
package org.stuartaroth.httpjava.services.id;

import java.util.UUID;

public class UniqueIdService implements IdService {
    @Override
    public String generateId() {
        return UUID.randomUUID().toString();
    }

    @Override
    public String generateId(String prefix) {
        return prefix + generateId();
    }
}

File: httpcsharp/httpcsharp/services/id/UniqueIdService.cs
using System;

namespace httpcsharp.services.id
{
    public class UniqueIdService : IIdService
    {
        public string GenerateId()
        {
            return Guid.NewGuid().ToString().Replace("-", "");
        }

        public string GenerateId(string prefix)
        {
            return prefix + GenerateId();
        }
    }
}